sql/plan/recursive_cte.go

```diff
@@ -46,12 +46,7 @@ const cteRecursionLimit = 1000
 // projection count and types. [Init] will be resolved before
 // [Rec] or [RecursiveCte] to share schema types.
 type RecursiveCte struct {
-	// non-recursive projection
-	Init sql.Node
-	// optionally recursive projection
-	Rec sql.Node
-	// if set, use a hashmap to skip tuples seen more than once
-	Deduplicate bool
+	*Union
```

Comment:
Embedding types like this is generally a bad idea that can introduce sneaky bugs through side effects of method inheritance over time. Unless you have a really good reason, prefer a named field

Reply:
i can change it, but we actually want an rCTE to be treated the same as a UNION after we validate and replace the recursive binding